criu/include/sigact.h

```diff
@@ -0,0 +1,14 @@
+#ifndef __CR_SIGACT_H__
+#define __CR_SIGACT_H__
+
+#include "images/core.pb-c.h"
+
+extern rt_sigaction_t sigchld_act;
+
+struct parasite_ctl;
+struct pstree_item;
+
+extern int prepare_sigactions(CoreEntry *core);
+extern int parasite_dump_sigacts_seized(struct parasite_ctl *ctl, struct pstree_item *);
```

Comment:
I don't feel like parasite_dump_sigacts_seized belongs here, why do we move it from criu/include/parasite-syscall.h? I believe this is not what we want.

Reply:
could you elaborate why we should not want to do that?

Reply:
> I don't feel like parasite_dump_sigacts_seized belongs here, why do we move it from criu/include/parasite-syscall.h? I believe this is not what we want.

It was explicitly stated on https://github.com/checkpoint-restore/criu/issues/336 that it should be seperated.

Reply:
> could you elaborate why we should not want to do that?

> It was explicitly stated on https://github.com/checkpoint-restore/criu/issues/336 that it should be seperated.

Pros for leaving it where it is:

- All "parasite_*" functions are now in one place (ok except for parasite_collect_aios and parasite_dump_pages_seized), common names you know
- All those functions trigger parasite execution (through compel), so basically doing the same thing, so it is nice to have them close by, this way if we fix one, we would remember to fix others
- We don't need to export parasite_ctl everywhere only in criu/parasite-syscall.c

I'm not sure that I have strong arguments =)

If we still decide to move it, when maybe let's also delete `#include "images/sa.pb-c.h"` from criu/parasite-syscall.c? As SaEntry becomes unused.

Reply:
> > could you elaborate why we should not want to do that?
> 
> > It was explicitly stated on #336 that it should be seperated.
> 
> Pros for leaving it where it is:
> 
>     * All "parasite_*" functions are now in one place (ok except for parasite_collect_aios and parasite_dump_pages_seized), common names you know
> 
>     * All those functions trigger parasite execution (through compel), so basically doing the same thing, so it is nice to have them close by, this way if we fix one, we would remember to fix others
> 
>     * We don't need to export parasite_ctl everywhere only in criu/parasite-syscall.c
> 
> 
> I'm not sure that I have strong arguments =)
> 
> If we still decide to move it, when maybe let's also delete `#include "images/sa.pb-c.h"` from criu/parasite-syscall.c? As SaEntry becomes unused.

@avagin Should we revert these changes?

Reply:
> * All "parasite_*" functions are now in one place (ok except for parasite_collect_aios and parasite_dump_pages_seized), common names you know

This patch moves all sigact code in one file;)

> * All those functions trigger parasite execution (through compel), so basically doing the same thing, so it is nice to have them close by, this way if we fix one, we would remember to fix others

The same is true about this patch.

> * We don't need to export parasite_ctl everywhere only in criu/parasite-syscall.c

We don't export parasite_ctl anywhere, do we?

> 
> I'm not sure that I have strong arguments =)

They are not;). I vote for merging this change.